CONTRIBUTING.md

```diff
@@ -0,0 +1,48 @@
+# Contributing to Stryker Dashboard
+
+This is the contribution guide for Stryker Dashboard. Great to have you here! Here are a few ways you can help make this project better.
+
+## Creating issues
+
+Do you have an idea for a feature or have you found a bug? Please create an issue so we can talk about it!
+
+## Adding new features
+
+New features are welcome! Either as requests or proposals.
+
+1. Please create an issue first, so we know what to expect from you.
+2. Create a fork on your github account.
+4. Please create or edit unit tests or integration tests.
+5. Run the tests and verify they pass.
+6. Update documentation.
+
+## Maintainers
+
+### Using Milestones
```

Comment:
I don't know if milestones are necessary. To me, it's unclear how milestones should be grouped, and which milestone should be added.

Reply:
I was thinking about this a bit more and I feel like it could be grouped more easy if we group the issues we want to fix before the next open-source council. It would make it more easy for me if I could see what we planned to do for the current quarter (Q1) and what will have to be done for the next quarter (Q2).

This ofcourse, is in ideal situations where everything gets finished in a single quarter. Though sometimes the implementation of it could go on for longer...

---

An option would be to use small increments for bug fixes and/or maintenance (e.g. `0.14.0` -> `0.14.1`). I don't think using milestones are necessary for these small changes. For the quarterly 'versions' mentioned above, we can increase the minor version (e.g. `0.14.x` -> `0.15.0`). For these versions we should create milestones (this is something I could do 😊).

It leaves the major version but I am not so sure what to do with that, maybe for changes so big that they require an intern? 😁

---

I would like to know your thoughts about this too, @hugo-vrijswijk 🙂

Reply:
I think it's good to have a roadmap, but it's rare those are actually followed 😅. Do you want to have strict quarterly releases? It goes against the "release early and often" philosophy a bit. For Stryker4s, there are times when I haven't been able to work on something as much as I wanted to, or am just excited about a feature I want to release quickly. Milestones can be limiting in both of those situations. 

Organizing features that are likely to be picked up soon might be easier with a "high-prio" tag, or a project board with a "preparing/investigating" column. All in all, it might be more complicated than needed, so my suggestion is to Keep It Simple (S). 

> It leaves the major version but I am not so sure what to do with that, maybe for changes so big that they require an intern? 😁

It's up to you when you decide the project is mature enough for a major 1.0 release 😃. Generally a 1.0 also implies fewer breaking changes and some stability. Though with a deployed application a "breaking change" is more difficult to define than a library

Reply:
@xandervedder I would strongly suggest to simply follow semver. Major versions are for breaking changes, minor versions are for non-breaking additions, patches for fixes. Doesn't need to be more complicated than that.

Reply:
Note that outside contributors cannot manage milestones, and cannot add issues to milestones. This is something maintainers should do.

Reply:
> Do you want to have strict quarterly releases?

Not necessarily, no. But it does make it easier for me to showcase what has been done and what will be done. Or at least, where we will focus our attention. I feel like right now it's more along the lines of "this is fun" or "this needs to be done". It would make more sense to prioritize some issues above others.

> Organizing features that are likely to be picked up soon might be easier with a "high-prio" tag, or a project board with a "preparing/investigating" column. All in all, it might be more complicated than needed, so my suggestion is to Keep It Simple (S).

I still need to think about what is more intuitive to use, maybe a project board is better. Prioritizing them from top to bottom could be something we could do.

> Note that outside contributors cannot manage milestones, and cannot add issues to milestones. This is something maintainers should do.

I agree, but this was always my intention 🙂.

---

I will think about this for a bit. [I will probably pick the simplest option.](https://en.wikipedia.org/wiki/KISS_principle)